networkx/algorithms/flow/utils.py

```diff
@@ -42,6 +42,9 @@ def _rewind(self):
         self._it = iter(self._edges.items())
         self._curr = next(self._it)
 
+    def __eq__(self, other):
+        return self._edges == other._edges
```

Comment:
This is an interesting use of `__eq__`. The object is "CurrentEdge" which is a way to `get` the current edge during an iteration across all edges coming out of a given node. But equality is tested for by testing that the edge sets being iterated over are the same, not that the current edge is the same. 

Where are these objects compared? Is it in `assert_graphs_equal`?  It would be nice if we could compare the nodes from which these edges come, but I suspect that could be difficult. :)

Reply:
Yeah, I also have doubts about adding `__eq__` to such a stateful object, but doing so temporarily let me run more experiments and get more coverage. Happy to follow your lead for what to do here.

Reply:
> Where are these objects compared? Is it in `assert_graphs_equal`?

Yes.

Reply:
OK.... let's add an `__eq__` method. And when we check equality we can check both the state of all edges as well as the current edge.  I can justify this defn of equality more to myself based on the idea that the value of `get` and the population of values obtainable are the same for both. So the object quacks the same for both objects.  Does this pass all tests?  Other thoughts?
```suggestion
        return (self._curr, self._edges) == (other._curr, other._edges)
```

Reply:
From what I know about this class, that seems reasonable. However, `self._curr` isn't always defined, so two tests fail with your suggestion. I'm hesitant to change too much about a class that I'm unfamiliar with, so maybe the least invasive way to add `__eq__` would be to use e.g. `getattr(self, "_curr", None)` instead of `self._curr`. We could also set `self._curr = None` in `__init__`, but maybe that wasn't done for a reason.

Reply:
Nice description -- I did notice that `_curr` is not defined in `__init__` and wondered how fragile it was. Looks like somewhat fragile. The only case where `_curr` is not defined is when `_edges` is empty (or more precisely, when `if self._edges:` does not cause `rewind` to be called). 

So, I guess it is possible that someone somewhere is relying on the call to `get` raising based on no `_curr` attribute.  But that is a *very* remote possibility. Also, I am quite sure that leaving out setting it was an oversight and not intentional. But I'm also aware that calling `get()` on an empty edge set case will raise now -- and would return `None` if we set it. And I feel like raising is slightly better. I guess I really don't care either way -- but have a very slight preference for using `get(self, "_curr", None)` in `__eq__`. If you have a preference, do that...

Reply:
Thanks for your input @dschult! I updated to use `getattr` (my preference too), and tests pass :rocket: